src/github_comment.ts

```diff
@@ -13,9 +13,11 @@ export async function upsertGitHubComment(
   const testSuiteRunId = resp.result?.testSuiteRunId || '';
   const testSuiteName = resp.result?.testSuiteName || '';
   const results = resp.result?.results || [];
-  const failedTests = results.filter(x => x.success === false);
-  const successTests = results.filter(x => x.success === true);
-  const undefinedTests = results.filter(x => x.success === undefined);
+  const failedTests = results.filter(x => x.status === TestStatus.FAILED);
+  const successTests = results.filter(
+    x => x.status === TestStatus.PASSED || x.status === TestStatus.FLAKY
+  );
+  const undefinedTests = results.filter(x => x.status === TestStatus.ERROR);
```

Comment:
This works! If it's easier you can also checkout https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy -- but only if it's actually nicer to use, haha.

I think it'll work. Since node 21
![image](https://github.com/user-attachments/assets/5ce6eb49-eb9f-43fa-8b1b-8d5bb8f04cf4)

Reply:
Asked R1 to refactor it using `groupBy`:

```ts
  const groupedResults = groupBy(results, (x) => x.status);
  
  const failedTests = groupedResults[StatusEnum.FAILED] ?? [];
  const successTests = [
    ...(groupedResults[StatusEnum.PASSED] ?? []),
    ...(groupedResults[StatusEnum.FLAKY] ?? [])
  ];
  const undefinedTests = groupedResults[StatusEnum.ERROR] ?? [];
```

I don't think we gain anything in terms of readability. This change would only have some advantage if there were a lot of test results in the `results` array.